internal/statestore/redis.go

```diff
@@ -56,65 +56,126 @@ func (rb *redisBackend) Close() error {
 
 // newRedis creates a statestore.Service backed by Redis database.
 func newRedis(cfg config.View) Service {
-	// As per https://www.iana.org/assignments/uri-schemes/prov/redis
-	// redis://user:secret@localhost:6379/0?foo=bar&qux=baz
-
-	// Add redis user and password to connection url if they exist
-	redisURL := "redis://"
-	maskedURL := redisURL
+	return &redisBackend{
+		healthCheckPool: getRedisPool("healthcheck", cfg),
```

Comment:
the getRedisPool with a type "healthcheck" and "master" seems like a hack. furthermore, in their imlementation, you also skip to default case if not healthckeck. Are there other types of pools you envision using? If not, simply having 2 helpers getHealthCheckPool and getMasterPool might do? May be have a getSentinelPool for the common helper they both use? outside of that, most logic for these two seem separated into cases anyways.